llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -784,11 +784,11 @@ processInputModule(std::unique_ptr<Module> M) {
   Modified |= removeSYCLKernelsConstRefArray(*M.get());
 
   // There may be device_global variables kept alive in "llvm.compiler.used"
-  // to keep the optimizer from wrongfully removing them. Since it has served
-  // its purpose, these device_global variables can be removed. If they are not
-  // used inside the device code after they have been removed from
-  // "llvm.compiler.used" they can be erased safely.
-  Modified |= removeDeviceGlobalFromCompilerUsed(*M.get());
+  // to keep the optimizer from wrongfully removing them. llvm.compiler.used
```

Comment:
We intend to replace the translator with SPIR-V backend eventually. Can we remove the check at that point of time?

Thanks

Reply:
Once the SPIRV backend I think it's better to remove the `llvm.compiler.used` symbol in the backend, instead of in `sycl-post-link`, although some might disagree. I think it's OK to keep the current behaviour for the moment unless it becomes a problem later.